History/HistoryModel.cpp

```diff
@@ -380,53 +410,26 @@ void HistoryModel::scanInlineReferences()
         HistoryInlineRefs newRefs = refsById.value( e->id() );
         HistoryInlineRefs oldRefs = e->refs();
 
-        if( !newRefs.count() )
+        // sort newRefs: no need to sort oldRefs as it is already sorted)
+        std::sort( newRefs.begin(), newRefs.end(), lt );
```

Comment:
This sort thing is mega inefficient. Since std::sort cannot know anything about the used Qt container (which is the design, so it can be used by whatever container is compatible to the stl), it can only sort things in place.

However, there's two options:
1) We change the QList back to a QVector, then we'll basically have a C-style array of `HistoryInlineRef`, which is pretty memory effective (esp. since we don't modify that container once it is created). But sorting inside it means that the assignment operator of `HistoryInlineRef` has to be called 3 times for every swap. This can cause a lot of memory fragmentation. I'd rather avoid that for something that might be processed so frequently like this code.

2) We accept the change to QList. In that case, we're having one more indirection (QString + 64 bit most certainly doesn't fit into 64 bit). So, a `QList<HistoryInlineRef>` will essentially be a `QVector<HistoryInlineRef*>`. Adding to the first option, 2 or 3 `new` and `delete` calls for HistoryInlineRef, which has probably a size of 48 or 64 byte. Causing yet more memory fragmentation.

I'd assume that I was much more efficient to just sort the initial vector while creating it.